src/components/textfieldControl/js/Textfield.js

```diff
@@ -0,0 +1,138 @@
+
+    /*************
+     * Textfield *
+     *************/
+
+    /**
+     * A component for controlling a textfield and handling data binding.
+     * Typically this will be used in conjunction with a UI control widget such as
+     * button steppers or slider.
+     */
+    fluid.defaults("fluid.textfield", {
+        gradeNames: ["fluid.viewComponent"],
+        ariaOptions: {
+            // Specified by implementor
+            // ID of an external label to refer to with aria-labelledby
+            // attribute
+            // "aria-labelledby": ""
+        },
+        strings: {
+            // Specified by implementor
+            // text of label to apply to both textfield and slider input
+            // via aria-label attribute
+            // "aria-label": ""
+        },
+        modelListeners: {
+            value: {
+                "this": "{that}.container",
+                "method": "val",
+                args: ["{change}.value"]
+            }
+        },
+        listeners: {
+            "onCreate.bindChangeEvt": {
+                "this": "{that}.container",
+                "method": "change",
+                "args": ["{that}.setModel"]
+            },
+            "onCreate.initTextfieldAttributes": {
+                "this": "{that}.container",
+                method: "attr",
+                args: [{
+                    "aria-labelledby": "{that}.options.ariaOptions.aria-labelledby",
+                    "aria-label": "{that}.options.strings.aria-label"
+                }]
+            }
+        },
+        invokers: {
+            setModel: {
+                changePath: "value",
+                value: "{arguments}.0.target.value"
+            }
+        }
+    });
+
+    /**
+     * Sets the model value only if the new value is a valid number, and will reset the textfield to the current model
+     * value otherwise.
+     *
+     * @param {Object} that - the component
+     * @param {Number} value - the new numerical entry
+     * @param {String} path - the path into the model for which the value should be set
+     */
+    fluid.textfield.setModelRestrictToNumbers = function (that, value, path) {
+        var isNumber = !isNaN(Number(value));
+        if (isNumber) {
+            that.applier.change(path, value);
+        }
+
+        // Set the textfield to the latest valid entry.
+        // This handles both the cases where an invalid entry was provided, as well as cases where a valid number is
+        // rounded. In the case of rounded numbers this ensures that entering a number that rounds to the current
+        // set value, doesn't leave the textfield with the unrounded number present.
+        that.container.val(that.model.value);
+    };
+
+    /******************************
+     * TextField Range Controller *
+     ******************************/
+
+    /**
+     * Range Controller is intended to be used as a subcomponent of a fluid.textfield component. It will limit the input
```

Comment:
This came out of a previous [discussion](https://issues.fluidproject.org/browse/FLUID-5536?focusedCommentId=35623&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-35623) with @amb26. The idea is that the textfield should be general. Also users can interact with the textfield directly, while having something else manage the restrictions on the textfield. Unfortunately to be able to more easily setup the configuration I had to make the controller a parent grade instead of only a subcomponent.